src/extension.ts

```diff
@@ -0,0 +1,230 @@
+      outputChannel.appendLine(`[INFO] Skipped ${result.filesSkipped.length} agent(s) (already up to date):`);      
+    }
+    
+    if (result.errors.length > 0) {
+      outputChannel.appendLine(`[ERROR] Encountered ${result.errors.length} error(s) during installation:`);
+      for (const error of result.errors) {
+        outputChannel.appendLine(`[ERROR]   ${error}`);
+      }
+      
+      // Show error notification with action to view output
+      const action = await vscode.window.showErrorMessage(
+        'PAW agent installation encountered errors. Check output channel for details.',
+        'View Output'
+      );
+      
+      if (action === 'View Output') {
+        outputChannel.show();
+      }
+      
+      // Log remediation guidance
+      outputChannel.appendLine('[INFO] If you encounter permission errors, you can set a custom prompts directory:');
+      outputChannel.appendLine('[INFO]   1. Open Settings (Cmd/Ctrl+,)');
+      outputChannel.appendLine('[INFO]   2. Search for "paw.promptDirectory"');
+      outputChannel.appendLine('[INFO]   3. Set a path where you have write permissions');
+    } else {
+      // Full success - show success notification
+      const agentCount = result.filesInstalled.length;
+      vscode.window.showInformationMessage(
+        `PAW agents installed successfully. ${agentCount} agent${agentCount !== 1 ? 's' : ''} available in Copilot Chat.`
+      );
+      outputChannel.appendLine(`[INFO] PAW agent installation completed successfully`);
+    }
+  } catch (error) {
+    // Catch-all for unexpected errors during the installation check/process
+    const message = error instanceof Error ? error.message : String(error);
+    outputChannel.appendLine(`[ERROR] Failed to install PAW agents: ${message}`);
+    
+    const action = await vscode.window.showErrorMessage(
+      'PAW agent installation failed. Extension will continue without agents.',
+      'View Output'
+    );
+    
+    if (action === 'View Output') {
+      outputChannel.show();
+    }
+    
+    // Don't throw - allow extension to continue activating
+  }
+}
+
+/**
+ * Extension deactivation function called when extension is deactivated.
+ * 
+ * This function is called when the extension is deactivated, either because
+ * VS Code is shutting down or the extension is being disabled/uninstalled.
+ * 
+ * All resources registered via context.subscriptions are automatically disposed,
+ * so no manual cleanup is required in this implementation.
+ */
+export async function deactivate(): Promise<void> {
+  if (!sharedExtensionContext) {
+    return;
+  }
+
+  const outputChannel = sharedOutputChannel ?? vscode.window.createOutputChannel(OUTPUT_CHANNEL_NAME);
+  const createdTempChannel = sharedOutputChannel === undefined;
+
+  outputChannel.appendLine('[INFO] PAW Workflow extension deactivating - removing installed agents.');
+
+  try {
+    const cleanupResult = await removeInstalledAgents(sharedExtensionContext);
```

Comment:
We are removing agent files when the extension is deactivated - this is bad. The agents should ONLY be deleted if the extension is being uninstalled. Removing on deactivation has very bad effects when multiple vs code instances are open.